javaee8/security-1.0/src/main/java/org/wildfly/swarm/ts/javaee8/security/customformauth/CustomIdentityStore.java

```diff
@@ -0,0 +1,45 @@
+package org.wildfly.swarm.ts.javaee8.security.customformauth;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.security.enterprise.credential.UsernamePasswordCredential;
+import javax.security.enterprise.identitystore.CredentialValidationResult;
+import javax.security.enterprise.identitystore.IdentityStore;
+
+@ApplicationScoped
+public class CustomIdentityStore implements IdentityStore {
+    private static Map<String, String> emailsToPasswords;
+
+    private static Map<String, String> emailsToRoles;
+
+    static {
+        emailsToPasswords = new HashMap<String, String>() {
+            {
+                put("admin@mail.com", "adminPassword");
+                put("user@mail.com", "userPassword");
+            }
+        };
+        emailsToRoles = new HashMap<String, String>() {
+            {
+                put("admin@mail.com", "ADMIN");
+                put("user@mail.com", "USER");
+            }
+        };
+    }
+
+    public CredentialValidationResult validate(UsernamePasswordCredential credential) {
+        String caller = credential.getCaller();
+        if (emailsToPasswords.containsKey(caller) &&
+                emailsToPasswords.get(caller).equals(credential.getPasswordAsString())) {
+
+            return new CredentialValidationResult(caller.substring(0, caller.indexOf('@')),
+                                                  new HashSet<>(Arrays.asList(emailsToRoles.get(caller))));
```

Comment:
`emailsToRoles` should be `Map<String, List<String>>` IMHO, if we need a list of roles.

Reply:
Or, when using guava, it can be a `MultiMap<String, String>`, but the former is probably more explicit.